benches/nexmark/generator/person_generator.rs

```diff
@@ -0,0 +1,246 @@
+        name: next_person_name(rng),
+        email_address: next_email(rng),
+        credit_card: next_credit_card(rng),
+        city: next_us_city(rng),
+        state: next_us_state(rng),
+        date_time: DateTime::from_timestamp(secs.try_into().unwrap(), nsecs.try_into().unwrap()),
+        extra: String::new(),
+    }
+}
+
+/// Return a random person id (base 0).
+// TODO(absoludity): Update to take GeneratorConfig.
+pub fn next_base0_person_id<R: Rng + ?Sized>(event_id: Id, rng: &mut R) -> Id {
+    // Choose a random person from any of the 'active' people, plus a few 'leads'.
+    // By limiting to 'active' we ensure the density of bids or auctions per person
+    // does not decrease over time for long running jobs.
+    // TODO(absoludity): Understand why this code appears to shift the active
+    // people ids to always be the most recent people, rather than what the
+    // comment above claims.
+
+    // By choosing a person id ahead of the last valid person id we will make
+    // newPerson and newAuction events appear to have been swapped in time.
+    let num_people = last_base0_person_id(event_id) + 1;
+    let active_people = std::cmp::min(num_people, NUM_ACTIVE_PEOPLE);
+    let n = rng.gen_range(0..active_people);
+    num_people - active_people + n
+}
+
+/// Return the last valid person id (ignoring FIRST_PERSON_ID). Will be the
+/// current person id if due to generate a person.
+// TODO(absoludity): Update to take GeneratorConfig.
+pub fn last_base0_person_id(event_id: Id) -> Id {
+    let epoch = event_id / TOTAL_PROPORTION;
+    let mut offset = event_id % TOTAL_PROPORTION;
+
+    if offset >= PERSON_PROPORTION {
+        // About to generate an auction or bid.
+        // Go back to the last person generated in this epoch.
+        offset = PERSON_PROPORTION - 1;
+    }
+    // About to generate a person.
+    epoch * PERSON_PROPORTION + offset
+}
+
+// Return a random US state.
+fn next_us_state<R: Rng + ?Sized>(rng: &mut R) -> String {
+    US_STATES.choose(rng).unwrap().to_string()
+}
+
+// Return a random US city.
+fn next_us_city<R: Rng + ?Sized>(rng: &mut R) -> String {
+    US_CITIES.choose(rng).unwrap().to_string()
+}
+
+// Return a random person name.
+fn next_person_name<R: Rng + ?Sized>(rng: &mut R) -> String {
+    format!(
+        "{} {}",
+        FIRST_NAMES.choose(rng).unwrap(),
+        LAST_NAMES.choose(rng).unwrap()
+    )
+}
+
+// Return a random email address.
+fn next_email<R: Rng + ?Sized>(rng: &mut R) -> String {
+    format!("{}@{}.com", next_string(rng, 7), next_string(rng, 5))
+}
+
+// Return a random credit card number.
+fn next_credit_card<R: Rng + ?Sized>(rng: &mut R) -> String {
+    format!(
+        "{:04} {:04} {:04} {:04}",
+        rng.gen_range(0..10_000),
+        rng.gen_range(0..10_000),
+        rng.gen_range(0..10_000),
+        rng.gen_range(0..10_000)
+    )
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use chrono::NaiveDate;
+    use rand::rngs::mock::StepRng;
```

Comment:
More of a note than anything else, but for the real deal (not tests) we probably want to use something like [xoshiro](https://docs.rs/rand_xoshiro/) with a fixed seed for our rng source so that we get reasonably distributed data while still being deterministic

Reply:
Great - thanks, I'll take a look (and yes, I'm only using the mock `StepRng` here for deterministic unit tests - not random at all).

Reply:
Yep, that's totally fine for the tests since no one wants to have to figure out what values the rng will yield